pkg/rkt-stage1/0008-Set-working-directory-for-containers.patch

```diff
@@ -0,0 +1,87 @@
+From c43301a359d39273e669cb9f166217a791781705 Mon Sep 17 00:00:00 2001
+From: bharani-zededa <bharani@zededa.com>
+Date: Sun, 24 Nov 2019 15:02:57 -0800
+Subject: [PATCH] Set working directory for containers
+
+---
+ build.sh           |  2 +-
+ files/launcher.sh  | 12 ++++++++++++
+ files/run          |  4 ++++
+ kernel/init-initrd |  7 ++++++-
+ 4 files changed, 23 insertions(+), 2 deletions(-)
+ create mode 100644 files/launcher.sh
+
+diff --git a/build.sh b/build.sh
+index adeb817..04fceb6 100755
+--- a/build.sh
++++ b/build.sh
+@@ -24,7 +24,7 @@ then
+     exit 1
+ fi
+ 
+-execs="enter run stop udhcpc_script.sh"
++execs="enter run stop udhcpc_script.sh launcher.sh"
+ netplugins="main/ptp main/bridge main/macvlan main/ipvlan ipam/host-local meta/flannel meta/tuning"
+ 
+ # Clean the repo, but save the vendor area
+diff --git a/files/launcher.sh b/files/launcher.sh
+new file mode 100644
+index 0000000..2fad27e
+--- /dev/null
++++ b/files/launcher.sh
+@@ -0,0 +1,12 @@
++#!/bin/sh
++
++mkdir -p $WORKDIR
++cd $WORKDIR
++ARGS=
++for i in "$@"
++do
++    ARGS="$ARGS \"$i\""
++done
++
++set -- $ARGS
++eval $@
+diff --git a/files/run b/files/run
+index 482c0ef..0ce4fa2 100755
+--- a/files/run
++++ b/files/run
+@@ -93,6 +93,8 @@ else
+     fi
+ fi    
+ 
++cp $stage1/launcher.sh $stage2
++chmod +x $stage2/launcher.sh
+ echo $cmdline > $mountpoint/cmdline
+ export IFS=$'\n'
+ stage2manifest=$mountpoint/manifest
+@@ -102,6 +104,8 @@ for row in `cat $stage2manifest | jq -c .app.environment[]`; do
+   VALUE=$(echo $row | jq -r .value)
+   echo export $KEY=\'$VALUE\' >> $stage2env
+ done
++WORKDIR=`cat $stage2manifest | jq -r -c .app.workingDirectory`
++echo export WORKDIR=\'$WORKDIR\' >> $stage2env
+ 
+ xl create $gargs $outconfig $STAGE1_XL_OPTS
+ domname=$(sed -ne '/^name *=/s/^.*"\(.*\)"/\1/p' < "$outconfig")
+diff --git a/kernel/init-initrd b/kernel/init-initrd
+index 66d8b54..c00f05f 100755
+--- a/kernel/init-initrd
++++ b/kernel/init-initrd
+@@ -74,7 +74,12 @@ then
+     ip link set dev eth0 up
+     udhcpc --interface=eth0 --script=/mnt/udhcpc_script.sh
+ fi
++cp /mnt/cmdline /mnt/rootfs
++cp /mnt/launcher.sh /mnt/rootfs
++chmod +x /mnt/rootfs/launcher.sh
+ cmd=`cat /mnt/cmdline`
+ echo "Executing $cmd"
+ source /mnt/environment
+-eval chroot /mnt/rootfs $cmd
++#export -p
++eval chroot /mnt/rootfs /launcher.sh $cmd
++#eval chroot /mnt/rootfs /bin/sh
```

Comment:
Is this line left over from some debug session? Ditto for line 82?

Reply:
I left the debug code commented out so it is easier to enable it when needed. I will remove the commented lines and upload a new diff.